client.go

```diff
@@ -23,6 +23,7 @@ type Conn struct {
 	net.Conn                         // a net.Conn holding the connection
 	UDPSize        uint16            // minimum receive buffer for UDP messages
 	TsigSecret     map[string]string // secret(s) for Tsig map[<zonename>]<base64 secret>, zonename must be in canonical form (lowercase, fqdn, see RFC 4034 Section 6.2)
+	TsigProvider                     // An implementation of the TsigProvider interface.
```

Comment:
This should be `TsigProvider TsigProvider` so as not to embed it's methods in `Conn`.

Reply:
Also probably the comment should be something a little more detailed and specific.

Reply:
Updated.